core/trino-main/src/main/java/io/trino/metadata/TableHandle.java

```diff
@@ -28,16 +29,19 @@
     private final CatalogHandle catalogHandle;
     private final ConnectorTableHandle connectorHandle;
     private final ConnectorTransactionHandle transaction;
+    private final Identity executedAs;
```

Comment:
Can you please extract adding this field as separate commit, so it will be easy to notice how this is field is used.